### PR TITLE
Add purchase API

### PR DESCRIPTION
**Pull request by Gadzhieva**

### CODE REVIEW

File: src/cinema/CustomExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.web.server.ResponseStatusException;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class CustomExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(value = {ResponseStatusException.class})
+    protected ResponseEntity<ExceptionDto> handleConflict(ResponseStatusException ex, WebRequest request) {
+        return new ResponseEntity<>(new ExceptionDto(ex.getReason()), ex.getStatus());
+    }
```
Comment (GadzhievFarid) [inline src/cinema/CustomExceptionHandler.java]: Why do we use ResponseStatusException if we are returning custom ExceptionDto anyway? I think throwing custom exceptions makes more sense

File: src/cinema/Main.java [hunk 2]
```diff
@@ -4,14 +4,21 @@
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.context.annotation.Bean;
 
+import javax.swing.text.TabExpander;
```
Comment (GadzhievFarid) [inline src/cinema/Main.java]: This import is not used

File: src/cinema/Seat.java [hunk 3]
```diff
@@ -3,10 +3,17 @@
...
+    boolean available = true;
 
     public Seat(int row, int column) {
         this.row = row;
         this.column = column;
+        if (row <= 4) {
+            price = 10;
+        } else {
+            price = 8;
+        }
```
Comment (GadzhievFarid) [inline src/cinema/Seat.java]: Magic numbers should be extracted to constants with self-descriptive names

File: src/cinema/Seat.java [hunk 4]
```diff
@@ -3,10 +3,17 @@
 public class Seat {
     int row;
     int column;
+    int price;
+    boolean available = true;
```
Comment (GadzhievFarid) [inline src/cinema/Seat.java]: Fields should be private

File: src/cinema/SeatDto.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+    public int getPrice() {
+        return price;
+    }
+
+    public SeatDto(Seat seat) {
+        this.row = seat.getRow();
+        this.column = seat.getColumn();
+        this.price = seat.getPrice();
+    }
```
Comment (GadzhievFarid) [inline src/cinema/SeatDto.java:27]: Would be nice to have all constructors near each other

File: src/cinema/TheaterService.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                  "The number of a row or a column is out of bounds!");
+        }
+        Seat seat = getSeatByRowAndColumn(row, column);
+        if (seat.isAvailable()) {
+            seat.setAvailable(false);
+            return new SeatDto(seat);
+        } else {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
+                    "The ticket has been already purchased!");
+        }
```
Comment (GadzhievFarid) [inline src/cinema/TheaterService.java]: Minor: I think this is a bit more readable

```suggestion
        if (!seat.isAvailable()) {
            throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
                    "The ticket has been already purchased!");
            }

        seat.setAvailable(false);
        return new SeatDto(seat);
```

File: src/cinema/TheatreController.java [hunk 7]
```diff
@@ -1,17 +1,21 @@
 package cinema;
 
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 
 @RestController
 public class TheatreController {
```
Comment (GadzhievFarid) [inline src/cinema/TheatreController.java]: Is it Theater or Theatre? We should decide and stick to one spelling

File: src/cinema/Seat.java (hunk 3 above)
Comment (GadzhievFarid) [inline src/cinema/Seat.java]: `this` is used not uniformly, i.e. `this.column`, but just `price`

**Review by GadzhievFarid — CHANGES_REQUESTED** [review]

**Comment by Gadzhieva** [comment]
Thank you, fixed

**Review by GadzhievFarid — APPROVED** [review]
Good job

**Pull request merged**